src/rule_definition_tools.jl

```diff
@@ -88,8 +88,15 @@ macro scalar_rule(call, maybe_setup, partials...)
     )
     f = call.args[1]
 
-    frule_expr = scalar_frule_expr(__source__, f, call, setup_stmts, inputs, partials)
-    rrule_expr = scalar_rrule_expr(__source__, f, call, setup_stmts, inputs, partials)
+    # Generate variables to store derivatives named dfi/dxj
+    derivatives = map(keys(partials)) do i
+        syms = map(j -> esc(gensym(Symbol("df", i, "/dx", j))), keys(inputs))
```

Comment:
Isn't this really the same as:
```suggestion
        syms = map(j -> Symbol("df", i, "/dx", j)), keys(inputs))
```
Since if we don't escape, it will gensym on it's own.

Reply:
Ah, yes, that's a good point. And I agree on the rename, `∂f / ∂x` is a better name for partial derivatives